### PR TITLE
[REFACTOR] payments sse endpoint

### PR DESCRIPTION
**Pull request by jackstar12**
The sse endpoint doesnt work if there is no properly configured reverse proxy that takes care of disabling gzip. A small wrapper around `GzipMiddleware` can fix that tough.

The code for the endpoint itself seems a bit overcomplicated to me aswell (why an additional queue?), refactored that aswell.

### CODE REVIEW

File: lnbits/core/views/api.py [hunk 1]
```diff
@@ -421,34 +421,18 @@ async def subscribe_wallet_invoices(request: Request, wallet: Wallet):
     logger.debug(f"adding sse listener for wallet: {uid}")
     api_invoice_listeners[uid] = payment_queue
 
-    send_queue: asyncio.Queue[Tuple[str, Payment]] = asyncio.Queue(0)
-
-    async def payment_received() -> None:
-        while True:
-            try:
-                async with async_timeout.timeout(1):
```
Comment (jackstar12) [inline lnbits/core/views/api.py:429]: why was there a timeout previously? it seems like it did nothing at all

File: lnbits/core/models.py [hunk 2]
```diff
@@ -169,6 +169,8 @@ async def update_status(
     async def set_pending(self, pending: bool) -> None:
         from .crud import update_payment_status
 
+        self.pending = pending
```
Comment (jackstar12) [inline lnbits/core/models.py:172]: the function previously only updated the db, now updates the pending attribute aswell (see comment later)

File: lnbits/core/views/api.py [hunk 3]
```diff
@@ -421,34 +420,18 @@ async def subscribe_wallet_invoices(request: Request, wallet: Wallet):
...
-                    payment: Payment = await payment_queue.get()
-                    if payment.wallet_id == this_wallet_id:
-                        logger.debug("sse listener: payment received", payment)
-                        await send_queue.put(("payment-received", payment))
-            except asyncio.TimeoutError:
-                pass
-
-    task = asyncio.create_task(payment_received())
```
Comment (dni) [inline lnbits/core/views/api.py:437]: haha, what a useless task this was! good catch

File: lnbits/core/views/api.py [hunk 4]
```diff
@@ -421,34 +421,18 @@ async def subscribe_wallet_invoices(request: Request, wallet: Wallet):
...
-
     try:
         while True:
             if await request.is_disconnected():
                 await request.close()
                 break
-            typ, data = await send_queue.get()
-            if data:
-                jdata = json.dumps(dict(data.dict(), pending=False))
```
Comment (jackstar12) [inline lnbits/core/views/api.py:446]: because `set_pending` previously did not set pending, it was set in here. Not necessary anymore

**Review by dni — APPROVED** [review]

File: lnbits/core/views/api.py (hunk 3 above)
Comment (motorina0) [reply lnbits/core/views/api.py:437]: why useless?`

Comment (jackstar12) [reply lnbits/core/views/api.py:437]: what use did it have? It seems like it just added an extra queue values had to unnecessarily pass through

Comment (motorina0) [reply lnbits/core/views/api.py:437]: > what use did it have?
- I do not know
- but also I couldn't figure out where messages are added to `payment_queue` (or to `api_invoice_listeners`). I could only locate `get/pop/items` calls.

**Review by motorina0 — APPROVED** [review]

**Pull request merged**